src/refactor_nrepl/find/symbols_in_file.clj

```diff
@@ -63,24 +69,33 @@
          (let [rdr (-> path slurp core/file-content-sans-ns
                        readers/indexing-push-back-reader)
                rdr-opts {:read-cond :allow :features #{dialect} :eof :eof}
-               syms (atom #{})
-               collect-symbol (fn [form]
-                                ;; Regular symbol
-                                (when (symbol? form)
-                                  (swap! syms conj (normalize-ctor-call form)))
-                                ;; Classes used in typehints
-                                (when-let [t (:tag (meta form))]
-                                  (swap! syms conj t))
-                                (when (and (keyword? form)
-                                           (core/fully-qualified? form))
-                                  (swap! syms conj
-                                         (symbol (core/prefix form)
-                                                 (core/suffix form))))
-                                form)]
-           (loop [form (reader/read rdr-opts rdr)]
-             (when (not= form :eof)
-               (walk/prewalk collect-symbol form)
-               (recur (reader/read rdr-opts rdr))))
-           (->> @syms
-                (map (partial fix-ns-of-backquoted-symbols (dialect parsed-ns)))
-                set)))))))
+               syms (atom #{})]
+           (letfn [(collect-symbols [form]
+                     (let [minimal-meta (strip-meta-from-reader (meta form))]
+                       ;; metadata used in metadata-based protocol extensions
+                       ;; see #239 for an example
+                       (when minimal-meta
+                         (->> minimal-meta
+                              seq
+                              flatten
```

Comment:
`flatten` is somewhat fragile:

```
> (flatten #{1 2 3})
()
```

You call seq right before so my example can't happen, but still :)

So, I think you can simply call `(prewalk collect-symbols minimal-meta)` here

Reply:
This should be fine as it's just flattening a map via `seq`.

Didn't know `flatten` behaved like this.  Thanks for the heads up!